# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Realm.cpp
# Commit: c7eae2ddd714
# Full Hash: c7eae2ddd71487cbb16eeb32e3d0c347206cdd12
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-09-15 15:26:38
# Regressor Bug: 1730699
# File Overlap Count: 1
# Description:
#   Bug 1730699 part 1 - Treat empty iterators for null/undefined as immutable and unlinked. r=jonco
#   
#   This changes the empty iterator objects we create for for-in with null/undefined
#   to be immutable and unlinked.
#   
# ==============================================================================

diff -r 61c4f93eaf75 -r c7eae2ddd714 js/src/vm/Realm.cpp
--- a/js/src/vm/Realm.cpp	Wed Sep 15 08:52:32 2021 +0000
+++ b/js/src/vm/Realm.cpp	Wed Sep 15 09:20:44 2021 +0000
@@ -380,8 +380,7 @@
                                         "ObjectRealm::enumerators")) {
       ni->unlink();
     }
-    MOZ_ASSERT_IF(ni->objectBeingIterated(),
-                  &ObjectRealm::get(ni->objectBeingIterated()) == this);
+    MOZ_ASSERT(&ObjectRealm::get(ni->objectBeingIterated()) == this);
     ni = next;
   }
 }
